arch/rv32i/src/syscall.rs

```diff
@@ -125,7 +137,7 @@ impl kernel::syscall::UserspaceKernelBoundary for SysCall {
           //
           // ```
           // 34*4(sp):          <- original stack pointer
-          // 33*4(sp): syscall_args
+          // 33*4(sp):
```

Comment:
If we aren't storing this, then can we remove the space for it as well? Would need to decrement the addi here as well as in lib.rs.

Reply:
Theoretically yes.  Considering the delicate context, I wanted to keep the blast radius as small as possible for the moment.

Reply:
Is leaving this portion of the stack frame unchanged for now a deal-breaker?  Since the `syscall_args` local is gone, this still should be a net improvement in the stack usage of the function.

Reply:
Is this on your radar to update in the future? I'm just wary because of past experience, but if you are planning to update it in the future then I say full steam ahead on this PR.

Reply:
Yes, I'd like to do another clean-up pass once more of the dust settles from the OpenTitan bring-up effort.